### PR TITLE
If pgSettings is a function, evaluate it in createPostGraphQLHttpRequestHandler

### PR DESCRIPTION
**Pull request by explicitcall**
Please let me know if you're happy with the implementation, I'll proceed with the docs then.

### CODE REVIEW

File: src/postgraphql/postgraphql.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ type PostGraphQLOptions = {
   exportJsonSchemaPath?: string,
   exportGqlSchemaPath?: string,
   bodySizeLimit?: string,
-  pgSettings?: { [key: string]: mixed },
+  pgSettings?: { [key: string]: mixed } | ((req: Object) => Promise<Object>),
```
Comment (benjie) [inline src/postgraphql/postgraphql.ts]: Better type: `(req: IncomingMessage) => Promise<{[key: string]: mixed }>`

**Comment by benjie** [comment]
You can make this a PR to master then v3 can benefit (and I'll merge into v4)

**Comment by explicitcall** [comment]
resubmitted as #556

**Comment by benjie** [comment]
(Note for the future: you can edit a PR target branch without closing/reopening:

screenshot 2017-08-16 15 32 12


)

**Pull request closed without merging**